internal/controller/utils.go

```diff
@@ -251,6 +280,45 @@ func getVolumeForPDDeployment(ctx context.Context, msvc *msv1alpha1.ModelService
 	return volumes
 }
 
+// getEnvsForcontainer returns the desired list of env vars for the container for the given URI type
+// For hf URIs, it returns an EnvVar which has a reference to a secretKey, provided by ModelArtifacts,
+// with HF_TOKEN in that secret
+// Other URI types do not need the controller to add any EnvVars
+func getEnvsForcontainer(ctx context.Context, msvc *msv1alpha1.ModelService) []corev1.EnvVar {
+	envs := []corev1.EnvVar{}
+	var desiredEnv *corev1.EnvVar
+
+	uriType := UriType(msvc.Spec.ModelArtifacts.URI)
+
+	switch uriType {
+	// only HF case needs a EnvVar, PVC and OCI don't
+	case HF:
+		if msvc.Spec.ModelArtifacts.AuthSecretName != nil {
+			desiredEnv = &corev1.EnvVar{
+				Name: ENV_HF_TOKEN,
```

Comment:
I think we also need to create an `HF_HOME` environment variable to point to the volumeMount mountPath. From https://docs.vllm.ai/en/latest/models/supported_models.html:

> By default, vLLM loads models from [Hugging Face (HF) Hub](https://huggingface.co/models). To change the download path for models, you can set the HF_HOME environment variable; for more details, refer to [their official documentation](https://huggingface.co/docs/huggingface_hub/package_reference/environment_variables#hfhome).

Reply:
I see... that's vLLM specific. I'll include it, but the other approach is to have the user specify that in the baseconfig. We definitely need to document the consequence of URIs in the user guide.

Reply:
Our HF_HOME would take precedence

Reply:
Done.